x-pack/plugins/spaces/server/spaces_client/spaces_client.test.ts

```diff
@@ -117,11 +125,46 @@ describe('#getAll', () => {
     });
   });
 
+  test('strips solution property in serverless build', async () => {
+    const mockDebugLogger = createMockDebugLogger();
+    const [SOWithSolution] = savedObjects;
+    const mockCallWithRequestRepository = savedObjectsRepositoryMock.create();
+    mockCallWithRequestRepository.find.mockResolvedValue({
+      saved_objects: [SOWithSolution],
+    } as any);
+    const mockConfig = createMockConfig();
+
+    const client = new SpacesClient(
+      mockDebugLogger,
+      mockConfig,
+      mockCallWithRequestRepository,
+      [],
+      'serverless'
+    );
+    const [actualSpace] = await client.getAll();
+    const [{ solution, ...expectedSpace }] = expectedSpaces;
```

Comment:
TIL: I was unaware of grabbing the "rest" with a spread in the destructure! ++cool!